test/jdk/tools/jpackage/share/MultiLauncherTwoPhaseTest.java

```diff
@@ -0,0 +1,99 @@
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.nio.file.Path;
+import jdk.jpackage.test.AdditionalLauncher;
+import jdk.jpackage.test.PackageTest;
+import jdk.jpackage.test.PackageType;
+import jdk.jpackage.test.TKit;
+import jdk.jpackage.test.JPackageCommand;
+
+/**
+ * Test multiple launchers in two phases. First test creates app image and then
+ * creates installer from this image. Output of the test should be
+ * MultiLauncherTwoPhaseTest*.* installer. The output installer should be basic
+ * installer with 3 launcher MultiLauncherTwoPhaseTest, bar and foo. On Windows
+ * we should have start menu integration under MultiLauncherTwoPhaseTest and
+ * desktop shortcuts for all 3 launchers. Linux should also create shortcuts for
+ * all launchers.
+ */
+
+/*
+ * @test
+ * @summary Multiple launchers in two phases
+ * @library ../helpers
+ * @library /test/lib
+ * @key jpackagePlatformPackage
+ * @build jdk.jpackage.test.*
+ * @modules jdk.incubator.jpackage/jdk.incubator.jpackage.internal
+ * @run main/othervm/timeout=360 -Xmx512m MultiLauncherTwoPhaseTest
+ */
+
+public class MultiLauncherTwoPhaseTest {
+
+    public static void main(String[] args) throws Exception {
+        TKit.run(args, () -> {
+            Path appimageOutput = TKit.workDir().resolve("appimage");
+
+            JPackageCommand appImageCmd = JPackageCommand.helloAppImage()
+                    .setArgumentValue("--dest", appimageOutput);
+
+            AdditionalLauncher launcher1 = new AdditionalLauncher("bar");
+            launcher1.setDefaultArguments().applyTo(appImageCmd);
+
+            AdditionalLauncher launcher2 = new AdditionalLauncher("foo");
+            launcher2.applyTo(appImageCmd);
+
+            PackageTest packageTest = new PackageTest()
+                    .addRunOnceInitializer(() -> appImageCmd.execute())
+                    .addBundleDesktopIntegrationVerifier(true)
+                    .addInitializer(cmd -> {
+                        cmd.addArguments("--app-image", appImageCmd.outputBundle());
+                        cmd.removeArgumentWithValue("--input");
+                    })
+                    .forTypes(PackageType.WINDOWS)
+                    .addInitializer(cmd -> {
+                        cmd.addArguments("--win-shortcut", "--win-menu",
+                                "--win-menu-group", "MultiLauncherTwoPhaseTest");
+                    })
+                    .addInstallVerifier(cmd -> {
+                        launcher1.verify(cmd);
```

Comment:
There is no need in explicit call to AdditionalLauncher.verify() as this function is scheduled for the call from AdditionalLauncher.applyTo() call. 
Thus there is no need to change access to this function from "private" to "public" in your patch to AdditionalLauncher.java.